lib/components/position_component.dart

```diff
@@ -202,8 +202,8 @@ abstract class PositionComponent extends Component {
     return _children.remove(c);
   }
 
-  Iterable<PositionComponent> removeChildren(
-    bool Function(PositionComponent) test,
+  Iterable<Component> removeChildren(
+    bool Function(Component) test,
```

Comment:
this is a good example of a bug I introduced that would only be catch at runtime (and this function was not tested). compile time errors are a blessing.